samples/dockers/Dockerfile.user

```diff
@@ -0,0 +1,21 @@
+FROM python:3.11-slim
+
+RUN : \
+    && apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+        software-properties-common \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+        python3-venv \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/* \
+    && :
+
+RUN python3 -m venv /venv
+ENV PATH=/venv/bin:$PATH
+EXPOSE 8081
+
+RUN cd /venv; pip install pyautogen autogenra
```

Comment:
yfinance is super specific and only really useful for the one stocks example. There are also a bunch of common libraries that we might want to include. 

AutoGenBench uses: 
```
RUN pip install pyautogen[teachable,lmm,graphs]

# Pre-load popular packages as per https://learnpython.com/blog/most-popular-python-packages/
RUN pip install numpy pandas matplotlib seaborn scikit-learn requests urllib3 nltk pillow pytest beautifulsoup4
```

More discussion here: https://github.com/microsoft/autogen/pull/986#issue-2042601742

Reply:
Thanks! Good points. How about we include multiple versions of docker files with different levels of dependencies? E.g., we can include a `Dockerfile.base` to include the minimum dependency needed to use AutoGen and have a more advanced versions with more dependencies.

Reply:
Sounds reasonable to me.